projects/gloo/api/v1/settings.proto

```
@@ -385,6 +385,26 @@ message Settings {
 
         // Options to configure Gloo's integration with [Kubernetes](https://www.kubernetes.io/).
         GrafanaIntegration grafanaIntegration = 1;
+
+        // Provides the ability to specify which label(s) to apply to the
+        // `*_config_status` metric for each resource type.
+        message ConfigStatusMetricsOptions {
+            // Defines which labels to apply for a metric.
+            message MetricLabels {
+                // Each (key, value) pair in the map defines a label to be applied.
+                //   key:   Specifies the name of the label (i.e. "namespace")
+                //   value: Specifies the jsonpath string corresponding to the field on the resource (i.e. "metadata.namespace")
+                map<string, string> label_to_path = 1;
+            }
+
+            // Labels for the virtual_service_config_status metric. If unspecified
+            // or the map is empty, then the virtual_service_config_status metric
+            // will not be recorded.
+            MetricLabels virtual_service_labels = 1;
```

any reason for the nested message instead of just using a `map<string, string>` here? what kinds of things are we anticipating adding here?

---

Yes, we'll be adding `MetricLabels gateway_service_labels = 2`, `MetricLabels route_table_labels = 3`, and so on

---

Could we instead have a type that defines the GVK for a resource (https://github.com/solo-io/gloo/blob/master/install/helm/gloo/crds/gateway.solo.io_v1_VirtualService.yaml#L9) instead of having the explcit types mentioend in the API?

---

We could. Is there a way to represent that as a single string? I feel like this yaml is going to real messy real quick

This would also open the door for users to specify a resource type that we don't validate, and therefore can't record these metrics on. That's easy enough to detect though, so we could fail loudly in that case